retrofit_gen/lib/parser/parser.dart

```diff
@@ -7,9 +7,8 @@ import '../parsed_info/parsed_info.dart';
 Req _parseReq(String httpMethod, DartObject annot, MethodElement method) {
   final reader = ConstantReader(annot);
   String path = reader.read('path').stringValue;
-  final metaDataContent = reader.read('metaData');
-  Map<DartObject, DartObject> metaData =
-      metaDataContent.isNull ? null : metaDataContent.mapValue;
+  Map<String, String> metadata = reader.read('metadata').mapValue.map(
```

Comment:
Why putting everything as string and loose the real type ?? If you want to have `"needAuth": true` you'll get `"needAuth": "true"` which I personally don't like if it's possible to have it as bool directly

Reply:
Bool is encoded into a string value of bool by https://github.com/Jaguar-dart/client/pull/19/files/d4aefbf49862c1adbb32ed84e58f4395838b11b5#diff-876682d5c26d60b3e34e5223251f0017R139. So all values including Lists and maps are supposed to be preserved as is. If there is a bug in the implementation of `toStringValue` please raise an issue.

Reply:
Yes I have seen the toStringValue but if I understand correctly I'll have at the end `"needAuth": "true"` under my interceptor and not `"needAuth": true` under the example I print the metadata, I'll get your branch and check this